src/libutil/file-system.cc

```diff
@@ -21,8 +21,14 @@ namespace fs = std::filesystem;
 
 namespace nix {
 
-Path absPath(Path path, std::optional<PathView> dir, bool resolveSymlinks)
+Path absPath(PathView path, std::optional<PathView> dir, bool resolveSymlinks)
 {
+    // If we `path` to refer to a newly-created string, we set this
+    // variable to that string and then set `path` to this. This ensures
```

Comment:
nit: I think you meant something like this?

```suggestion
    // If we set `path` to refer to a newly-created string, we set
    // `scratch` to that string first, and then set `path` to `scratch`. This ensures
```